### DISCUSSION TITLE
How to get started writing a custom plugin for Parcel

### DISCUSSION THREAD

**Discussion by BenjaminWaye** [parcel-bundler/parcel#6353] [Q&A] [answered]
Is there any material on how to get started writing a custom plugin for Parcel? 

Followed https://v2.parceljs.org/plugin-system/resolver/

But parcel seems to limit what can be imported in the config. It doesn't seem to be possible to write a path to the plugin, writing:
```
{
    "extends": "@parcel/config-default",
    "resolvers": ["...", ".parcel/parcel-resolver-url.js"]
}
``` 
I get the error message that parcel plugins need to be named "parcel-resolver-name"

Context: I am trying to rebase the url paths of image assets in css files to dist. Like from: "../../dir/dir/file.png" to: "./dist/file.png"

**Comment by mischnic** [comment, marked as answer]
You have to specify a package name there, see https://github.com/parcel-bundler/parcel/issues/4321

**Reply by BenjaminWaye** [reply]
By using @mischnic repo https://github.com/mischnic/parcel-resolver-root it is possibly to quickly get a testable custom resolver up and running by simply replacing the code. Thanks @mischnic !
